src/components/modals/TransactionModal/TransactionModal.tsx

```diff
@@ -0,0 +1,231 @@
+  interactionType,
+  contractId,
+  // todo: implement advanced settings
+  // todo: implement smartweave favorites
+  state,
+  showModal,
+}: {
+  // main control for how the modal is build
+  transactionType: TRANSACTION_TYPES;
+  interactionType?: ANT_INTERACTION_TYPES;
+  state?: AntMetadata;
+  contractId?: ArweaveTransactionID; // contract ID if asset type is a contract interaction
+  showModal: () => void;
+}) {
+  const [{ arnsSourceContract }] = useGlobalState();
+  const isMobile = useIsMobile();
+  const [accepted, setAccepted] = useState<boolean>(false);
+  const [toAddress, setToAddress] = useState<string>('');
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(false);
+
+  // todo: add "transfer to another account" dropdown
+
+  function getAssociatedNames(txId: ArweaveTransactionID) {
+    return Object.entries(arnsSourceContract.records)
+      .map(([name, id]) => {
+        if (id === txId.toString()) return name;
+      })
+      .filter((n) => !!n);
+  }
+
+  return (
+    <>
+      <div
+        className="modal-container"
+        style={isMobile ? { padding: 'none' } : {}}
+      >
+        {/**modal header */}
+        <div
+          className="flex flex-column card"
+          style={
+            isMobile
+              ? {
+                  width: '95%',
+                  maxWidth: 'none',
+                  height: '95%',
+                  padding: 0,
+                  boxSizing: 'border-box',
+                  gap: '1em',
+                }
+              : {
+                  width: 400,
+                  minHeight: 200,
+                  padding: 0,
+                  paddingBottom: 15,
+                  gap: '1em',
+                }
+          }
+        >
+          <div
+            className="flex flex-column flex-space-between"
+            style={{
+              borderBottom: '2px rgb(0,0,0,.2) solid',
+              padding: 15,
+              boxSizing: 'border-box',
+              marginTop: 5,
+              gap: '.5em',
+              position: 'relative',
+            }}
+          >
+            <span className="text-medium white bold">
+              {transactionType}({interactionType})&nbsp;:&nbsp;
+              {state?.state.ticker}
+            </span>
+            <span className="flex faded text" style={{ alignItems: 'center' }}>
+              Contract ID:&nbsp;
+              <CopyTextButton
+                copyText={contractId ? contractId.toString() : ''}
+                displayText={`${contractId!
+                  .toString()
+                  .slice(0, isMobile ? 6 : 15)}...${contractId!
+                  .toString()
+                  .slice(isMobile ? -6 : -15)}`}
+                size={25}
+                wrapperStyle={{
+                  fontStyle: 'bold',
+                  justifyContent: 'center',
+                  alignItems: 'center',
+                }}
+                position="relative"
+              />
```

Comment:
text is too big 

Figma
<img width="757" alt="image" src="https://user-images.githubusercontent.com/12800001/218200565-5b1deb4d-5835-4b6d-9030-9dbb8591050f.png">

Reply:
fixed